codec-http2/src/main/java/io/netty/handler/codec/http2/StreamBufferingEncoder.java

```diff
@@ -118,7 +131,9 @@ public StreamBufferingEncoder(Http2ConnectionEncoder delegate, int initialMaxCon
 
             @Override
             public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData) {
-                cancelGoAwayStreams(lastStreamId, errorCode, debugData);
+                goAwayDetail = new GoAwayDetail(
+                    lastStreamId, errorCode, ByteBufUtil.getBytes(debugData));
```

Comment:
please add a comment why its safe to use `getBytes(...)` here... something like this: 

```suggestion
                    // Using getBytes(...) is safe here as GoAwayDetail(...) will clone the byte[].
                    lastStreamId, errorCode, ByteBufUtil.getBytes(debugData));
```